src/Razor/src/Microsoft.AspNetCore.Razor.ProjectEngineHost/Serialization/CompilationTagHelperResolver.cs

```diff
@@ -0,0 +1,73 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the MIT license. See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Generic;
+using System.Collections.Immutable;
+using System.Diagnostics;
+using System.Linq;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.AspNetCore.Razor.Language;
+using Microsoft.AspNetCore.Razor.Telemetry;
+using Microsoft.CodeAnalysis;
+using Microsoft.CodeAnalysis.Razor;
+
+namespace Microsoft.AspNetCore.Razor.ProjectEngineHost.Serialization;
+
+internal class CompilationTagHelperResolver
```

Comment:
This code comes from the "DefaultTagHelperResolver", but as well as performing tag helper discovery from a compilation, that class also was MEF exported into Roslyn, and was in the middle of an inheritance chain. Again just dependencies that aren't always relevant or desirable.